src/tls/key_exchange.h

```diff
@@ -3,37 +3,30 @@
 #pragma once
 
 #include "crypto/entropy.h"
+#include "crypto/key_pair.h"
 #include "crypto/mbedtls/key_pair.h"
 #include "ds/logger.h"
 #include "tls/error_string.h"
 
 #include <iostream>
 #include <map>
 #include <mbedtls/ecdh.h>
+#include <mbedtls/ecp.h>
+#include <mbedtls/pk.h>
 
 namespace tls
 {
   class KeyExchangeContext
   {
   private:
     crypto::mbedtls::ECDHContext ctx = nullptr;
-    std::vector<uint8_t> own_public;
-    crypto::EntropyPtr entropy;
+    std::vector<uint8_t> key_share, peer_key_share;
```

Comment:
Minor: I'd prefer the definition on two separate lines as it's more consistent with the existing code base, and less prone to initialisation errors.